tests/Network/MethodsTest.cpp

```diff
@@ -147,6 +147,58 @@ TEST_F(NetworkMethodsTest, request_failure) {
   ASSERT_FALSE(res.hasResponse());
 }
 
+TEST_F(NetworkMethodsTest, request_failure_on_status_not_acceptable) {
```

Comment:
We need these `request_failure-*` tests on sendRequestRetry as well.
I think they should behave the same, but the code-path differs.